aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -177,10 +184,18 @@ module aptos_framework::fungible_asset {
                 project_uri,
             }
         );
-        move_to(metadata_object_signer, Supply {
-            current: 0,
-            maximum: maximum_supply
-        });
+
+        if (features::concurrent_fungible_asset_enabled()) {
+            move_to(metadata_object_signer, ConcurrentSupply {
+                current: aggregator_v2::create_aggregator(option::get_with_default(&mut maximum_supply, MAX_U128)),
```

Comment:
Shouldn't we still enforce the maximum the user specifies here? It can be a lot smaller than MAX_U128

Reply:
aggregator enforces the maximum - create_aggregator has max_value argument. I pass there maximum_supply if set, or MAX_U128 if maximum_supply is None

Reply:
Gotcha is it easy to add unit tests for different cases here?

Reply:
@movekevin what is the use case for having `Option<u128>` for max supply? It is either a real MAX_U128 in case of none, or a user-define value also capped at MAX_U128, I do not see why would we ever want to differentiate between the two apart from having an error code and a few more instructions to check. From gas costs perspective this also does not save much